engine/CMakeLists.txt

```diff
@@ -1539,3 +1550,33 @@ ENDIF (WIN32 AND NOT UNIX)
 
 INCLUDE(CPack)
 
+INCLUDE(FetchContent)
+FetchContent_Declare(
+    googletest
+    URL https://github.com/google/googletest/archive/609281088cfefc76f9d0ce82e1ff6c30cc3591e5.zip
+    )
+# For Windows: Prevent overriding the parent project's compiler/linker settings
+SET(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
+FetchContent_MakeAvailable(googletest)
```

Comment:
Since we'll probably still need to support CMake version 3.11, this line will probably need to be changed to fit the pattern:

```cmake
FetchContent_GetProperties(googletest)
if(NOT googletest_POPULATED)
  FetchContent_Populate(googletest)
  ...
endif()
```

See https://cmake.org/cmake/help/latest/module/FetchContent.html#command:fetchcontent_getproperties

Reply:
I think this is the only change remaining to be made, other than merging in the latest from master after #536 gets merged. Then we should be good to go.

Reply:
... and #536 has now been merged into master. @royfalk those changes should be ready for you to merge into this branch, whenever you get the chance.